packages/react-storage/src/components/StorageBrowser/context/actions/listLocationsAction.ts

```diff
@@ -61,21 +61,23 @@ export const createListLocationsAction = (
     let nextNextToken: ListLocationsOutput['nextToken'] = refresh
       ? undefined
       : nextToken;
-    let remainingPageSize = pageSize;
 
     do {
-      remainingPageSize = remainingPageSize - locationsResult.length;
+      const remainingPageSize = pageSize - locationsResult.length;
```

Comment:
- simplified logic: HowManyItemsDoWeWant (constant) - HowManyItemsDoWeHave
- `pageSize` is actually the input pageSize which is constant
- Computing a new `remainingPageSize` instead of carrying it from the previous run